universe/assets_3d/api_assets_3d.go

```diff
@@ -0,0 +1,189 @@
+package assets_3d
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/momentum-xyz/ubercontroller/universe"
+	"github.com/momentum-xyz/ubercontroller/universe/api"
+	"github.com/momentum-xyz/ubercontroller/universe/api/dto"
+	"github.com/momentum-xyz/ubercontroller/utils"
+)
+
+func (a *Assets3d) apiGetAssets3d(c *gin.Context) {
+	queryParams := struct {
+		kind string `form:"kind" json:"kind"`
+	}{}
+
+	if err := c.ShouldBind(&queryParams); err != nil {
+		err = errors.WithMessage(err, "Assets3d: apiGetAssets3d: failed to bind query parameters")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_query", err, a.log)
+	}
+
+	a3dMap := make(map[uuid.UUID]universe.Asset3d)
+	predicateFn := func(asset3dID uuid.UUID, asset3d universe.Asset3d) bool {
+		entry := asset3d.GetEntry()
+		kind := utils.GetFromAnyMap(entry.Meta, "kind", "")
+		return kind == queryParams.kind
+	}
+
+	if queryParams.kind == "" {
+		a3dMap = a.GetAssets3d()
+	} else {
+		a3dMap = a.FilterAssets3d(predicateFn)
+	}
+
+	assets := make([]*dto.Asset3d, 0, len(a3dMap))
+
+	for _, el := range a3dMap {
+		asset := el.GetEntry()
+
+		assetDTO := &dto.Asset3d{
+			ID:        asset.Asset3dID.String(),
+			CreatedAt: asset.CreatedAt.String(),
+			UpdatedAt: asset.UpdatedAt.String(),
+		}
+		assets = append(assets, assetDTO)
+	}
+
+	c.JSON(http.StatusOK, assets)
+}
+
+func (a *Assets3d) apiAddAssets3d(c *gin.Context) {
+	inQuery := struct {
+		assets3dIDs []string `form:"assets3d_ids[]" binding:"required"`
```

Comment:
still not sure, is it ok or they should provide object with id and meta, but if you don't know too, let's keep it as it is now